app/controllers/loyalty_program_data_controller.rb

```diff
@@ -3,11 +3,13 @@ class LoyaltyProgramDataController < ApplicationController
 
   # GET /loyalty_program_data or /loyalty_program_data.json
   def index
-    @loyalty_program_data = LoyaltyProgramDatum.all
-    @transaction = Transaction.all
+    @loyalty_program_data = LoyaltyProgramDatum.where(account_id: current_user)
```

Comment:
Did you test this out? seems like there might be error here. 

 should it be current_user.account.id ?